src/common/common.actions.js

```diff
@@ -0,0 +1,35 @@
+import * as constants from './common.actionTypes'
+
+export const getGraphQL = (payload, variables, onSuccess, onError) => {
+    onSuccess = onSuccess || ((a) => a)
+    return (dispatch) => {
+        onError = onError || ((a) => dispatch(apologize(a)))
+        return new Promise((resolve) => {
+            let request = new XMLHttpRequest()
```

Comment:
Shouls we use a lib that support promises from the get go?  Just to cut down on the boilerplate ;)
I used this one in the past: https://github.com/mzabriskie/axios